src/keycloak/keycloak_openid.py

```diff
@@ -581,6 +581,25 @@ def introspect(self, token, rpt=None, token_type_hint=None):
             )
         return raise_error_from_response(data_raw, KeycloakPostError)
 
+    @staticmethod
+    def _verify_token(token, key: Union[str, jwk.JWK, jwk.JWKSet, None], **kwargs):
+        # keep the function free of IO
+        # this way it can be used by `decode_token` and `a_decode_token`
+        if isinstance(key, str):
+            key = "-----BEGIN PUBLIC KEY-----\n" + key + "\n-----END PUBLIC KEY-----"
+            key = jwk.JWK.from_pem(key.encode("utf-8"))
+
+        if key is not None:
```

Comment:
I believe this is a breaking change. Previously you could specify a key and set `validate=False`, the key would have no effect and the decoded token would be returned. Now if I specify a key, validation is always done no matter what the `validate` value is. I'd suggest that the validation should only be performed if `validate=True`

Reply:
Oh, good catch! I was too focused on the `validate=True`...

I thought the same, but actually the behavior is different on `master` - the validation will still happen. This is because `kwargs` will contain `key`, which will cause `jwcrypto` to run the validation. So `master` has a bug. This is an example fix for `master`
```python
        if validate:
            # ...
        else:
            kwargs.pop("key", None) # have to remove the key
            full_jwt = jwt.JWT(jwt=token, **kwargs)
            full_jwt.token.objects["valid"] = True
            return json.loads(full_jwt.token.payload.decode("utf-8"))
```

I can fix the bug in this PR

Reply:
Fixed, also added some unit tests to cover it. Let me know what you think

Reply:
Nice catch as well! :) thanks a lot!